### PR TITLE
Vinai/dolt commit author no config

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
Add a bats test that models the following behavior.

1. Unsets name and user. 
2. Makes a sql change
3. Add a commit with --author

### CODE REVIEW

File: bats/config.bats [hunk 1]
```diff
@@ -131,3 +131,32 @@ teardown() {
...
+     dolt config --global --unset user.email
+
+     run dolt config --list
+     [ "$status" -eq 0 ]
+     [ "${#lines[@]}" -eq 0 ]
+
+      dolt add .
+      run dolt commit --author="John Doe <john@doe.com>" -m="works"
+      [ "$status" -eq 0 ]
+
```
Comment (andrew-wm-arthur) [inline bats/config.bats:161]: can you run `dolt log` and assert the author was written correctly?

Comment (VinaiRachakonda) [reply bats/config.bats:165]: Added!

Comment (andrew-wm-arthur) [reply bats/config.bats:165]: This asserts that message was written correctly, but we also need to check that the author info was written correctly

Comment (VinaiRachakonda) [reply bats/config.bats:161]: Whoops! Think I got it now?

Comment (andrew-wm-arthur) [reply bats/config.bats:161]: Ideally you'd check both, but this looks right

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM

**Pull request merged**